lib/charms/layer/nginx.py

```diff
@@ -53,7 +69,6 @@ def configure_site(site, template, **kwargs):
     hookenv.log('Wrote vhost config {} to {}'.format(context, template),
                 'info')
 
-    if os.path.exists('/etc/nginx/sites-enabled/default'):
-        os.remove('/etc/nginx/sites-enabled/default')
-    host.service_reload('nginx')
```

Comment:
I'm confused. So there was already code in `configure_site` to remove the default site. The kubeapi-load-balancer charm calls `configure_site` here: [link](https://github.com/juju-solutions/kubernetes/blob/2c44da01b1a825718ea9adbdfaa7599bd0ed333c/cluster/juju/layers/kubeapi-load-balancer/reactive/load_balancer.py#L148)

So shouldn't the default site have been removed already?

Reply:
I think the issue is that this happens too late; the layer has already errored and blocked

Reply:
Where does the error come from? I would like to know where that's occurring -- otherwise, it's hard to judge if moving this to the `nginx_ready` handler will really fix the issue.

To be honest, I suspect a different cause: the `host.service_reload('nginx')`. If nginx is immediately failing after the apt package is installed, then it won't be running by the time `host.service_reload` is called. And a "reload" does not restart a failing service:

```
$ sudo service nginx stop
$ sudo service nginx reload
nginx.service is not active, cannot reload.
```

So it could be the case that the default site _does_ get removed, but the nginx service never gets restarted.

It would need to be `host.service_restart('nginx')` or `host.service_reload('nginx', restart_on_failure=True)` ([link](https://github.com/juju/charm-helpers/blob/cd5c5154f32fd7c92ee6cffda52cf696be2b7bf2/charmhelpers/core/host.py#L141)) instead, I think.

Have you been able to reproduce the issue? If not, I would recommend finding a way to do that, or seeking clarification from the issue reporter on the symptoms so we can better understand what's really going on.

Reply:
I've added the `restart_on_failure` parameter as I can't reproduce this, even with removing IPv6 from a container.  I do, however, see the benefit in removing the default site at install time.  Let me know what you think.

Reply:
If we can't reproduce then I'm good with a speculative fix. Thanks for looking into it.